### DISCUSSION TITLE
In the c++ client implementation,Why is object data acquired from the metadta?

### DISCUSSION THREAD

**Discussion by liusitan** [v6d-io/v6d#967] [General]
Describe your problem
---------------------
In the c++ client, get_objet is based on get_metadata, and the object is acquired from the buffer section of the metadata, which 
contradicts the statement in the document that: 

> Vineyard represents all kinds of data as vineyard objects, for any vineyard object, it consists of two parts:
> 
> A set of blobs where the payload of the data lives in;
> 
> A hierarchical meta tree that describes the type, layout, and properties of the data.
> 
> The decoupling design of data payload and data layout brings three benefits:
>
it suggests that the metadata and payload are stored separately,
----------------------------------------------------------------------

metadata and payload are still stored in different places. Although in the client-side, object are acquired from the metadata, they are stored in a separate place in the server, metadata are stored in the metastored, and payloads are stored through the Apache-Array.

**Comment by sighingnow** [comment]
> which contradicts the statement in the document that:

Actually `get` includes two part: `get_metadata()`, then `get_buffers()` that involves in the returned metadata.

**Comment by sighingnow** [comment]
Closing as answered.
